packages/@sanity/migrate/src/mutations/types.ts

```diff
@@ -2,12 +2,18 @@ import type {Path} from '@sanity/types'
 import type {Operation} from './operations/types'
 import {Optional} from './typeUtils'
 
+/**
+ * A list of {@link NodePatch} objects.
+ */
 export type NodePatchList =
   | [NodePatch, ...NodePatch[]]
   | NodePatch[]
   | readonly NodePatch[]
   | readonly [NodePatch, ...NodePatch[]]
 
+/**
+ * A sanity document
+ */
 export type SanityDocument = {
```

Comment:
Don't we have this type elsewhere in the Sanity code base? Won't it always have `_updatedAt` and `_createdAt` as well?

Reply:
Somehow missed this comment.

We do have it defined in quite a few places already, but there's not a huge of value in re-using it IMO as long as the definitions are structurally compatible. This type is mostly used with `create({_type: 'foo'})` et. al., so making `_updatedAt`, `_createdAt` etc. mandatory would require you to specify those when creating new documents from a migration.

However, I agree these fields could/should be defined as optional strings.